src/main/java/org/commcare/core/sandbox/SandboxUtils.java

```diff
@@ -65,9 +68,16 @@ public static FormInstance loadFixture(UserSandbox sandbox,
             }
         }
 
+        if (appFixtureStorage != null) {
+            return loadAppFixture(appFixtureStorage, refId, userId);
+        }
         return loadAppFixture(sandbox, refId, userId);
     }
 
+    public static FormInstance loadFixture(UserSandbox sandbox, String refId, String userId) {
```

Comment:
Why do the other places that call this method directly not also need to try to retrieve the app-level fixture storage first?

Reply:
Hmm, both of those usages have the comment `//TODO: Wow. This is.... kind of megasketch` so I'm guessing they're pretty sketch